examples/webpack-demo-vanilla-bundle/src/examples/example07.ts

```diff
@@ -315,6 +315,76 @@ export class Example7 {
     this.sgb.filterService.clearFilters();
   }
 
+  
+  allFilters() {
+    const grid = this.sgb;
+    const modalHtml:string =`<div id="modal-allFilter" class="modal is-active" >
+        <style type="text/css">
+            .table {
+                display: table;
+            }
+
+            .row {
+                display: table-row;
+            }
+
+            .column {
+                display: table-cell;
+                vertical-align: top;
+                width: 40%;
+            }
+        </style>
+        <div class="modal-background"></div>
+        <div class="modal-card">
+            <header class="modal-card-head">
+                <p class="modal-card-title">Filter</p>
+                <button class="delete btn-close" aria-label="close"></button>
+            </header>
+            <section class="modal-card-body">
+              <div class="slickgrid-container grid-pane">
+                <div id="modal-allFilter-content">
+                  <div id="modal-allFilter-table" class="table slick-headerrow ui-state-default">
+                  </div>
+                </div>
+              </div>
+            </section>
+            <footer class="modal-card-foot">
+              <button class="button btn-close">Close</button>
+              <button id="btn-clear-all" class="button">Clear All Filter</button>
+              <button class="button btn-close is-success">Search</button>
+            </footer>
+        </div>
+    </div>`;
+    
+    document.body.appendChild(DOMPurify.sanitize(modalHtml,{ RETURN_DOM: true }));
+
+    $(".btn-close").on('click',function() {
+      grid?.filterService.toggleHeaderFilterRow();
+      document.getElementById("modal-allFilter").remove();
+    });
+    $("#btn-clear-all").on('click',function() {
+      grid?.filterService.toggleHeaderFilterRow();
+      document.getElementById("modal-allFilter").remove();
+      grid?.filterService.clearFilters();
+    });
+
+    for (const columnFilter of grid?.columnDefinitions){
+      if (columnFilter.filterable){
+        const filterElm = `modal-allfilter-${columnFilter.id}`;
+        $('#modal-allFilter-table').append(`
+        <div class="row slick-headerrow-columns">
+          <div class="column">${columnFilter.name}</div><div id="${filterElm}" class="column ui-state-default slick-headerrow-column"></div>
+        </div>
+        `);
+        grid?.filterService.drawFilterTemplate(columnFilter,`#${filterElm}`);
+      }
+    }
+    if (grid?.slickGrid.getOptions().showHeaderRow){
+      grid?.filterService.toggleHeaderFilterRow();
```

Comment:
I was testing the UI and I could see the Filter bar being toggled (hidden) in the background because of this line, are there any particular reason you're doing this? or is it mostly for your use case? I'm just curious to know if it's related to how it works or not.

Reply:
Yes, there is a reason for this, synchronization of the two filter inputs (modal/title column).
When I type a filter in the modal, for performance reasons, I do not update the column filter, if in this case it was visible the same filter field would have different values, the update (synchronization) only occurs after closing the modal when it is displayed again the filter line.

Reply:
oh ok I see and that works because looking at the code I can see that calling `showHeaderRow` will recreate all the columns and their filters because it triggers this `onHeaderRowCellRendered` event which re-renders the filters. That's probably the only option you have then... or actually I think if you call `this.sgb.showHeaderRow(true)` it would probably work too without having to toggle the bar at all, you can give that a try and see if it makes any difference

Reply:
so I gave this a try and it works, probably better than toggling, I replaced all 3 toggle code with `this.sgb?.showHeaderRow(true);` (you'll have to replace `function` with fat arrow `() =>`. 

The end result is the same, calling the toggle or the `showHeaderRow` will both call a re-render of all filters after closing the modal which will keep filter values in sync. So did you want to give that a try or do you want me to merge as it is?